generator/src/util.rs

```diff
@@ -30,29 +30,30 @@ pub struct Split<'a, 'b> {
 impl<'a, 'b> Iterator for Split<'a, 'b> {
     type Item = &'a str;
     fn next(&mut self) -> Option<&'a str> {
-        if self.done {
-            return None;
-        } else if self.s.is_empty() {
-            self.done = true;
-            return Some("");
-        } else if !self.pending.is_empty() {
-            return Some(std::mem::take(&mut self.pending));
-        }
-
-        for (i, b) in self.s.bytes().enumerate() {
-            if b == b' ' || self.splitters.contains(&b) {
-                let ret = &self.s[..i];
-                // dont include the space, but include everything else on the next step
-                if b != b' ' {
-                    self.pending = &self.s[i..i + 1]
+        match (self.done, self.s.is_empty(), self.pending.is_empty()) {
+            (true, _, _) => None,
+            (_, true, _) => {
+                self.done = true;
+                Some("")
+            }
+            (_, _, false) => Some(std::mem::take(&mut self.pending)),
+            _ => {
+                for (i, b) in self.s.bytes().enumerate() {
+                    if b == b' ' || self.splitters.contains(&b) {
+                        let ret = &self.s[..i];
+                        // dont include the space, but include everything else on the next step
+                        if b != b' ' {
+                            self.pending = &self.s[i..i + 1]
+                        }
+                        self.s = &self.s[i + 1..];
+                        return Some(ret);
+                    }
                 }
-                self.s = &self.s[i + 1..];
-                return Some(ret);
+                // trailing data
+                self.done = true;
+                Some(self.s)
             }
         }
-        // trailing data
-        self.done = true;
-        Some(self.s)
     }
```

Comment:
I'm not convinced this is more readable than the existing code. Could you revert the change?

Reply:
Would you prefer to `else` as well?
If i take them out the first part would look like this
```
        if self.done {
            return None;
        } 
        if self.s.is_empty() {
            self.done = true;
            return Some("");
        } 
...
```
The reason I ask is that if-else with returns in them triggers some lints.

Of course if you'd rather keep them in, regardless, I'll do that.

Reply:
Those sound like quite opinionated lints then. I'm usually fond of functional style, but this is a bit too much IMO.